docs/design-documents/hal/0004-pin-names-general-guidelines.md

```diff
@@ -0,0 +1,132 @@
+# General Pin Names design document
+
+# Table of contents
+
+1. [Table of contents](#table-of-contents).
+    1. [Revision history](#revision-history).
+1. [Introduction](#introduction).
+    1. [Overview and assumptions](#overview-and-assumptions)
+1. [Detailed design](#detailed-design).
+    1. [LEDs](#leds).
+    1. [Buttons](#buttons).
+1. [Non-valid definitions](#non-valid-definitions).
+1. [Testing compliance](#testing-compliance).
+
+
+### Revision history
+
+1.0 - Initial revision - Malavika Sajikumar, Marcelo Salazar - August 2020  
+This document is written for Mbed OS 6.
+
+## Introduction
+
+### Overview and assumptions
+
+Mbed OS is designed so that application code written in the platform is portable across different Mbed supported boards with the same hardware capabilities or interfaces. However, the code may not be truly portable due to the differences in pin name definitions for the same kind of interfaces across different boards. 
+
+This document provides specific guidelines and best practices for defining pin names that apply to boards that implement and are compliant wit Arduino Uno compliant connector.
+
+
+## Detailed design
+
+To achieve meaningful portability of application code across various Mbed Enabled boards, certain pin names of commonly used interfaces and board components should be common across these boards. This document describes a set of rules on how to name generic pins in the board support package, specifically in PinNames.h file of the board.
+
+### LEDs
+
+**Definition of LEDs**
+
+All valid LED definitions should be defined as follows in PinNames.h:
+
+    // Px_xx relates to the processor pin connected to the LED
+    
+    #define LED1 = Px_xx  // LED1
+    #define LED2 = Px_xx  // LED2  
+    #define LED3 = Px_xx  // LED3  
+    #define LED4 = Px_xx  // LED4  
+    .  
```

Comment:
Important to clarify that it's not required to add 4 LEDs, only the ones available on the board.

Reply:
Done